ndb/src/google/cloud/ndb/model.py

```diff
@@ -1407,10 +1409,212 @@ def __init__(self, *args, **kwargs):
         raise NotImplementedError
 
 
+class _CompressedValue:
+    """A marker object wrapping compressed values.
+
+    Args:
+        z_val (bytes): A return value of ``zlib.compress``.
+    """
+
+    __slots__ = ("z_val",)
+
+    def __init__(self, z_val):
+        if not isinstance(z_val, bytes):
+            raise TypeError("Only bytes accepted", z_val)
```

Comment:
This was an `assert` in the original

Reply:
It looks like this check should be unnecessary, given _CompressedValue is private and assuming we trust the output of `zlib.compress`. Doesn't hurt, though.

Reply:
Yeah that's a good point and agrees with the concept of an `assert` over a full-fledged exception. I am :+1: on just removing this type check if you are?

Reply:
:+1: